### PR TITLE
Bundle the extension

### PR DESCRIPTION
**Pull request by itowlson**
Bundling velocitates extensioning.  By quite a bit.

**NOTE:** To get this to webpack, I had to rev a couple of packages that were bringing in unpackable packages.  This caused some minor knock-on effects in the application code, e.g. in port-forward.

**THINGS I STILL NEED TO DOUBLE CHECK:**

* ~~Telemetry (`aiKey` being loaded correctly)~~ Confirmed
* ~~Completion driven off the `snippets` directory~~ Confirmed

### CODE REVIEW

**Comment by itowlson** [comment]
+ @lstocchi for moar eyes on this beast

File: src/yaml-support/yaml-constant.ts [hunk 1]
```diff
@@ -8,10 +8,10 @@ export const VSCODE_YAML_EXTENSION_ID = 'redhat.vscode-yaml';
 
 export const KUBERNETES_SCHEMA_VERSION = '1.12.2';
 
-export const KUBERNETES_SCHEMA_FILE = path.join(__dirname, `../../../schema/swagger-v${KUBERNETES_SCHEMA_VERSION}.json`);
-export const FALLBACK_SCHEMA_FILE = path.join(__dirname, `../../../schema/swagger-v${KUBERNETES_SCHEMA_VERSION}.json`);
+export function KUBERNETES_SCHEMA_FILE(context: ExtensionContext) { return context.asAbsolutePath(`schema/swagger-v${KUBERNETES_SCHEMA_VERSION}.json`); }
+export function FALLBACK_SCHEMA_FILE(context: ExtensionContext) { return context.asAbsolutePath(`schema/swagger-v${KUBERNETES_SCHEMA_VERSION}.json`); }
 
-export const KUBERNETES_SCHEMA_ENUM_FILE = path.join(__dirname, `../../../schema/schema_enums-v${KUBERNETES_SCHEMA_VERSION}.json`);
+export function KUBERNETES_SCHEMA_ENUM_FILE(context: ExtensionContext) { return context.asAbsolutePath(`schema/schema_enums-v${KUBERNETES_SCHEMA_VERSION}.json`); }
```
Comment (lstocchi) [inline src/yaml-support/yaml-constant.ts:14]: In all three functions you could use the `assetPath` function and so avoid passing the context everytime, no?

Comment (itowlson) [reply src/yaml-support/yaml-constant.ts:14]: I think you're right.  This is most likely a historical accident where these changes date from the Before Giving Up On Avoiding Global State era of the refactor (where `assetPath` was my admission of defeat).  But it's just possible there might be some subtle ordering dependency where these functions could run before `setAssetContext` got called (due to code running during imports) - I don't remember such a thing but will double check, and I'll make the change you recommend unless I can find a good reason for the difference.  Thanks!

Comment (itowlson) [reply src/yaml-support/yaml-constant.ts]: Changed - thanks!

**Comment by gorkem** [comment]
Is this fixing #542 ?

**Comment by itowlson** [comment]
@gorkem Yes - bundling dramatically improves startup speed, and reduces CPU load and download size.

**Comment by gorkem** [comment]
@bnookala this looks like it is waiting on your review. 

@itowlson can we move this along, it is an important improvement

**Review by bnookala — APPROVED** [review]
long overdue +1, but also we should probably get that merge conflict fixed

**Comment by gorkem** [comment]
🎉 🎉 🎉

**Pull request merged**